lib/purl_fetcher/client/direct_upload_request.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require "digest"
+
+module PurlFetcher
+  class Client
+    DirectUploadRequest = Data.define(:checksum, :byte_size, :content_type, :filename) do
+      def self.from_file(path, file_name:, content_type:)
+        checksum = Digest::MD5.file(path).base64digest
```

Comment:
Is it worth letting the checksum to be passed in to avoid the overhead of regenerating?